### PR TITLE
Added ToolchainEvent CRD

### PR DESCRIPTION
**Pull request by sbryzak**
## Description
Add ToolchainEvent CRD to api.

Fixes https://issues.redhat.com/browse/CRT-1299

## Checks
1. Have you run `make generate` target? **[yes]**

2. Does `make generate` change anything in other projects (host-operator, member-operator)? **[yes]**

3. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/536

### CODE REVIEW

File: api/v1alpha1/toolchainevent_types.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+// ToolchainEventStatus defines the observed state of ToolchainEvent
+// +k8s:openapi-gen=true
+type ToolchainEventStatus struct {
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// Conditions is an array of current ToolchainEventStatus conditions
+	// Supported condition types:
+	// Complete
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply api/v1alpha1/toolchainevent_types.go]: You're right, it should have been Ready.  I've added the condition type and reason constants.

File: api/v1alpha1/toolchainevent_types.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	// The tier to assign to users registering for the event.  This must be the valid name of an nstemplatetier resource.
+	// +optional
+	Tier string `json:"tier,omitempty"`
+
+	// The unique activation code for the event
+	ActivationCode string `json:"activationCode"`
+
+	// If true, best effort is made to provision all attendees of the event on the same cluster
+	// +optional
+	PreferSameCluster bool `json:"preferSameCluster"`
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply api/v1alpha1/toolchainevent_types.go]: Fixed

**Comment by sbryzak** [comment]
I've also added VerificationRequired, we did discuss this in the design doc and it doesn't hurt to include it now.

**Review by alexeykazakov — APPROVED** [review]

**Comment by MatousJobanek** [comment]
Sorry for my late review.

@sbryzak It looks good, just don't forget to add the `toolchainevents` here: https://github.com/codeready-toolchain/toolchain-cicd/blob/master/scripts/add-cluster.sh#L61-L71

I'm letting you know just for the case you missed the warning https://github.com/codeready-toolchain/api/blob/master/make/generate.mk#L7

**Pull request merged**